logic/LogicManager.java

```diff
@@ -103,6 +106,14 @@ private FunctionalIterator<Rule> rulesWithNegations() {
         return rules().filter(rule -> !rule.when().negations().isEmpty());
     }
 
+    public FunctionalIterator<Unifier> unifiers(Concludable concludable, Rule rule) {
+        return iterate(applicableRules(concludable).getOrDefault(rule, new HashSet<>()));
+    }
+
+    public Map<Rule, Set<Unifier>> applicableRules(Concludable concludable) {
```

Comment:
add boolean `cache` which you can use to disable caching of concludables when it's a root query concludable, which could otherwise blow up the cache over time.

Reply:
Strange: 
At present, a new LogicCache object is created for each transaction (strange because it can be accessed through Transaction.session.database.borrowCache)